### DISCUSSION TITLE
dev question about  using streamlink API getting link

### DISCUSSION THREAD

**Discussion by CN-CODEGOD** [streamlink/streamlink#5934] [Q&A] [answered]
### Description

hello ! i am trying to write a powershell moduel to store streaming and play stream with streamlink , and i having trouble getting using the API of streamlink for my moduel
it work well when i use it in the shell ,but when i try to run the API in .py file the error occured 
![image](https://github.com/streamlink/streamlink/assets/166476136/8e63579d-8b20-4dba-b0f3-531011bb352f)

i am struggle to figure out why

**Comment by bastimeyer** [comment, marked as answer]
> AttributeError: partially initialized module 'streamlink' has no attribute 'streams' (most likely due to a circular import)

You already got your answer here:
https://github.com/streamlink/streamlink/issues/5927#issuecomment-2045096972

Rename your own `streamlink.py` file to something else, otherwise you can't import the `streamlink` module/package, because the `streamlink` name in Python's import system will already be taken by your module.

**Reply by CN-CODEGOD** [reply]
hello , thanks for the answer

**Reply by CN-CODEGOD** [reply]
can i use streamlink to check which stream is live ? if it can, i will be helpful for my project

**Reply by bastimeyer** [reply]
> can i use streamlink to check which stream is live

Streamlink is not really designed for this. It's a really poor choice to use Streamlink for checking whether a particular stream is online/available.

If you're talking about Twitch for example (which I'm sure you do, because you sent me a support question email a few days ago), then use their API instead of Streamlink:
https://dev.twitch.tv/docs/

**Reply by CN-CODEGOD** [reply]
because of the streamlink can check the  playable stream , so i was trying to use it to check the available. with it

**Reply by CN-CODEGOD** [reply]
can i use the stream url for the stream after i get the url ?

**Reply by bastimeyer** [reply]
> can i use the stream url for the stream after i get the url ?

Depends on the stream's expiration time and other parameters
